02.calendar/cal.rb

```diff
@@ -0,0 +1,48 @@
+require "date"
+require "optparse"
+
+opt = OptionParser.new
+opt.on("-y [VAL]") {|year| @year = year}
+opt.on("-m [VAL]") {|month| @month = month}
+opt.parse(ARGV)
+
+if @year != nil
+  @year
+else
+  @year = Date.today.year
+end
+if @month != nil
+  @month
+else
+  @month = Date.today.month
+end
+
+puts "#{@month}月 #{@year}".center(20)
+
+dayweeks = ["日","月","火","水","木","金","土"]
+dayweeks.each do |dayweek|
+  print "#{dayweek} "
+end
+puts "\n"
```

Comment:
```suggestion
puts dayweeks.join(" ")
```

`Array#join` で指定した文字を使って配列を文字列に連結することができます。
こっちの方が短く書ける、かつ良く使う処理なので、頭の隅にいれておいてください。